### DISCUSSION TITLE
Search Results WebPart not showing correct result for few string for People

### DISCUSSION THREAD

**Discussion by tusharprithviraj** [microsoft-search/pnp-modern-search#2714] [Q&A] [answered]
hi Team,

I am using pnp modern search web part 3.15.3.1 and for SearchBox i am using 4.6.1.0.

i have created custom template for people and binding data to it. Everything is working but for few search keyword it is showing unwanted profile.

for eg: if i search My name "Tushar" it will show me 1200 user profiles but actual are just 50 users with same name .

how can i target the search results to first filter with FirstName,LastName or just to check the accountname and show those results and then it can search the same keyword in other properties.

i would also like to understand how the results are pulled when a search query is fired for people properties and the function is present in which service file. i checked but not able to get it. i checked the search function

**Comment by wobba** [comment, marked as answer]
You can limit search to specific fields by using a query template like this:

`FirstName:({searchTerms}) OR LastName:({searchTerms}) OR AccountName:({searchTerms})`

Where you query in the managed properties with the data you are looking for,

**Reply by tusharprithviraj** [reply]
Thanks it worked...

**Reply by wobba** [reply]
Great :)
